airflow/models/taskinstance.py

```diff
@@ -205,19 +206,25 @@ def clear_task_instances(
         for job in session.query(BaseJob).filter(BaseJob.id.in_(job_ids)).all():  # noqa
             job.state = State.SHUTDOWN
 
-    if activate_dag_runs and tis:
+    if (dag_run_state is not False) and tis:
```

Comment:
```suggestion
    if dag_run_state is not False and tis:
```

I don’t think parantheses are needed here? If this feels ambiguous, maybe swap them?

```python
    if tis and dag_run_state is not False:
```